### ISSUE TITLE
set vim.wo.foldenable=false by default

### ISSUE THREAD

**Issue by cathaysia** [neovim/neovim#30027]
### Problem

foldenable is true by default, this makes code folded when open file. This make very strangle. Maybe we should set it to false by default.

### Expected behavior

foldenable is set to false by default.

**Comment by justinmk** [comment]
Defaults are changed when there is a clear case to be made for doing so. Do most users want folds or not? Don't see this in https://github.com/neovim/neovim/issues/21342 , maybe @echasnovski can comment

**Comment by echasnovski** [comment]
> Defaults are changed when there is a clear case to be made for doing so. Do most users want folds or not? Don't see this in #21342 , maybe @echasnovski can comment

The [raw data](https://gist.github.com/echasnovski/fa70dc75c475369747d2a485a13303fb#file-neovim-survey-summary_02_all-csv) (which can be queried in the top search bar) shows that default value of `foldenable=true` had about 86% of respondents (196 out of 227). My guess would be that this describes two sets of people: who actually use folds and who don't know about folds (and did not encounter problems with the default `foldmethod=manual` which use about 59% of respondents).

I personally love using folds with `foldmethod=indent` (which is quite universal and works in at least 90% of cases).

**Comment by justinmk** [comment]
> editors that provide folding functions, such as vscode, will display a folding button, and the code will be folded when the user clicks the button.

Then IMO this is blocked until we make the fold icon clickable.

**Comment by cathaysia** [comment]
vim uses shortcut keys instead of buttons to fold code, maybe we don't need to wait for the fold button

**Comment by zeertzjq** [comment]
> > editors that provide folding functions, such as vscode, will display a folding button, and the code will be folded when the user clicks the button.
> 
> Then IMO this is blocked until we make the fold icon clickable.

It's already clickable.

**Comment by cathaysia** [comment]
Closing this issue because there has been no non-automation comment for 686 days.

**Comment by justinmk** [comment]
Well, making the [fold icon clickable](https://github.com/neovim/neovim/issues/30027#issuecomment-2283841983) is still an option. But yes, some issues won't get activity for a long time. That doesn't necessarily mean they need comments, they can stay open to track ideas.

**Comment by cathaysia** [comment]
This issue was automatically closed by a [bot](https://github.com/cathaysia/cathaysia/actions/runs/28430527800/job/84243863606). I generally dislike tracking issues/PRs that have been open for too long :(.

If no progress can be made on the issue in the short term, I think the best approach might be to open a new one. Otherwise, the bot will keep closing it periodically.

**Comment by justinmk** [comment]
perhaps the bot should present the issues to you and you can make the decision, rather than leaving "not-stale" comments in project issues trackers?

there is nothing fundamentally wrong with old issues. some ideas take ages to resolve, but that doesn't mean that pretending they don't exist solves anything.

**Comment by cathaysia** [comment]
Yes, but in reality, if an issue or PR makes no progress within the first month, it is highly unlikely to see any progress in the future.

While keeping it open might seem useful, it effectively just leaves behind an issue that will never be resolved.

**Comment by justinmk** [comment]
nonsense. there are numerous ~10-year-old issues that we resolved in the last few years. and soon we will resolve https://github.com/neovim/neovim/issues/1496 , for example.

the stale-close bots were a bad idea, favored by maintainers that don't know how to [maintain a backlog](https://sink.io/jmk/priority/).
